### PR TITLE
Copy utilized elliptic & bn.js external packages into bitcore-common

### PR DESCRIPTION
**Pull request by MichaelAJay**
### Description
Creates bitcore-common package with elliptic & bn.js internal implementations. Switches elliptic/bn.js usage in bitcore-lib to bitcore-common implementations, and fixes direct accessor issues in bitcore-lib. All tests pass.

Performance evaluation availabe and pertinent to direct accessor fix

### Changelog

* Create bitcore-common package and implement required functionality from elliptic & bn.js -- NOTE: currently uses ES6 export/import
* bitcore-common test (bn, curve - base, secp256k1, short, point, jpoint, arith-paths, endomorphic edge cases)
* remove elliptic & bn.js from bitcore-lib, add bitcore-common to bitcore-lib
* replace external bn.js import in bitcore-lib/lib/crypto/bn.js with bitcore-common implementation
* replace external elliptic import in bitcore-lib/lib/crypto/point.js with bitcore-common implementation
* Replace all direct point component access in bitcore-lib with getter usage - crypto/schnorr.js, privatekey.js, publickey.js, test/publickey.js
* Performance differences are not notable, with the exception of ECDSA recovery, which saw a 14% performance regression and high variability. Could be because of ES export/import change.

### Testing Notes
All bitcore-lib tests Node tests pass. 

<hr style="border-width:3px">

### Checklist
* [x] I have read [CONTRIBUTING.md](https://github.com/bitpay/bitcore/tree/master/CONTRIBUTING.md) and verified that this PR follows the guidelines and requirements outlined in it.

### CODE REVIEW

File: packages/bitcore-common/lib/curve/base.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+    var res =  this.point(bytes.slice(1, 1 + len),
+                          bytes.slice(1 + len, 1 + 2 * len));
+
+    return res;
+  } else if ((bytes[0] === 0x02 || bytes[0] === 0x03) &&
+              bytes.length - 1 === len) {
+    return this.pointFromX(bytes.slice(1, 1 + len), bytes[0] === 0x03);
+  }
+  throw new Error('Unknown point format');
```
[the remark this answers is not in the document]
Comment (MichaelAJay) [reply packages/bitcore-common/lib/curve/base.js:285]: @kajoseph The current implementation is in parity with elliptic's implementation, but this would be an improvement. I've strived to make as few actual changes as possible.

There are skipped tests that document this deficiency - search for `// Known deficiency: decodePoint() accepts off-curve 0x04, 0x06, and 0x07 inputs.`

The proposed change does enable those tests to pass.

File: packages/bitcore-lib/index.js [hunk 2]
```diff
@@ -63,11 +63,9 @@ bitcore.Unit = require('./lib/unit');
 
 // dependencies, subject to change
 bitcore.deps = {};
-bitcore.deps.bnjs = require('bn.js');
 bitcore.deps.bs58 = require('bs58');
 
 bitcore.deps.Buffer = Buffer;
-bitcore.deps.elliptic = require('elliptic');
 bitcore.deps._ = require('lodash');
```
[the remark this answers is not in the document]
Comment (MichaelAJay) [reply packages/bitcore-lib/index.js:69]: Packages importing bitcore-lib will now be responsible for importing bn.js & elliptic directly.

File: packages/bitcore-common/lib/curve/base.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kajoseph) [reply packages/bitcore-common/lib/curve/base.js:285]: I'm inclined to go ahead an add this validation. Though I did look to see where decodePoint is being used and I didn't see it being used anywhere. Is it even needed?
